gateway/src/main/kotlin/builder/LoginBuilder.kt

```diff
@@ -0,0 +1,21 @@
+package dev.kord.gateway.builder
+
+import dev.kord.common.entity.PresenceStatus
+import dev.kord.gateway.Intents
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+
+data class Shards(val totalShards: Int, val indices: Iterable<Int> = 0 until totalShards)
+
+class LoginBuilder {
+    var presence: PresenceBuilder.() -> Unit = { status = PresenceStatus.Online }
```

Comment:
What's the reason for this being a lambda? Why not keep a presence builder locally instead?